app/Http/Controllers/Store/RegistrationController.php

```diff
@@ -358,18 +356,51 @@ function ($child) {
 
     public function update(StoreUpdateRegistrationRequest $request)
     {
+        $amendedCarers = [];
+
         $user = $request->user();
 
-        // Create New Carers
-        // TODO: Alter request to pre-join the array?
-        $carers = array_map(
-            function ($carer) {
-                return new Carer(['name' => $carer]);
+        // Fetch Registration and Family
+        $registration = Registration::findOrFail($request->get('registration'));
+
+        // Update primary carer.
+        $carerInput = (array) $request->get("pri_carer");
+        $carerKey = key($carerInput);
+        $carer = Carer::find($carerKey);
+        if ($carer->name !== $carerInput[$carer->id]) {
+            $carer->name = $carerInput[$carer->id];
+            $amendedCarers[] = $carer;
+        }
+
+        // Find secondary carers id's in the DB
```

Comment:
Would there be any point in adding a flag to carer as Main? So we have a distinct group for each - and can ensure that we always have a Main carer?

Would save all the manipulation with array keys below.